app/main.py

```diff
@@ -2,5 +1,19 @@
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
```

Comment:
The `storage` dictionary is defined at the function level, which means it will be shared across all instances of the decorated function. This is fine if the decorator is used for functions that don't need separate caches. If separate caches are needed, consider moving `storage` inside the `wrapper` function.